File: ArraysAndHashing/LongestConsecutive.cpp
#include <vector>
#include <set>

using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> s(nums.begin(),nums.end());
        int maxSequence = 0;
        int auxSequence = 0;
        for(auto it = s.begin() ; it != s.end() ; it++)
        {
            if(s.find(*it-1) == s.end())
            {
                int start = *it;
                while(s.find(start) != s.end())
                {
                    start++;
                    auxSequence++;
                }
                if(auxSequence > maxSequence)
                    maxSequence = auxSequence;
                auxSequence = 0;
            }
        }
        return maxSequence;
    }
};

File: ArraysAndHashing/LongestConsecutive.py
class Solution(object):
    def longestConsecutive(self, nums):
        s = set(nums)
        maxSequ = 0
        for i in nums:
            if (i-1) not in s:
        length = 0
                while((i+length) in s):
                    length += 1
                maxSequ = max(maxSequ,length)
         
        return maxSequ